src/openrct2-ui/CursorRepository.cpp

```diff
@@ -40,9 +40,12 @@ void CursorRepository::SetCurrentCursor(CursorID cursorId)
 {
     if (_currentCursor != cursorId)
     {
-        SDL_Cursor* cursor = _scaledCursors.at(_currentCursorScale).GetScaledCursor(cursorId);
-        SDL_SetCursor(cursor);
-        _currentCursor = cursorId;
+        if (_currentCursorScale < _scaledCursors.size())
+        {
+            SDL_Cursor* cursor = _scaledCursors.at(_currentCursorScale).GetScaledCursor(cursorId);
```

Comment:
at throws on invalid access so since we have performed the bounds check at is not required. Alternatively you could catch the exception